survey-sdk/src/main/java/co/nimblehq/survey/sdk/SurveyApi.kt

```diff
@@ -0,0 +1,62 @@
+package co.nimblehq.survey.sdk
+
+import co.nimblehq.survey.sdk.api.AppService
+import co.nimblehq.survey.sdk.entity.SurveyEntity
+import co.nimblehq.survey.sdk.network.NetworkBuilder
+import moe.banana.jsonapi2.ArrayDocument
+import moe.banana.jsonapi2.ObjectDocument
+
+class SurveyApi : NetworkBuilder() {
+    private lateinit var service: AppService
+    private var version = "v1"
+
+    companion object {
+        private lateinit var instancce: SurveyApi
+        fun getInstance(): SurveyApi = instancce
+
+        fun setup(): SurveyApi {
+            instancce = SurveyApi()
```

Comment:
should we mark the default constructor of `SurveyApi` to be private?

Reply:
@luongvo Good catch, it should have private constructor.